core/src/main/scala/codesearch/core/util/Helper.scala

```diff
@@ -40,47 +40,30 @@ object Helper {
     IO(Source.fromFile(path, "UTF-8")).bracket(source => IO.pure(source.getLines.toList))(source => IO(source.close()))
 
   def preciseMatch(query: String): String = {
-    val queryTokens: Seq[Token] = Tokenizer.parseStringWithSpecialSymbols(query)
+    val queryTokens: Seq[Token]  = Tokenizer.parseStringWithSpecialSymbols(query)
+    val specialChars: Set[String] = Set("|", "^", "$", "$", "+", "*", "(", ")", ".", "?")
     val preciseMatch: Seq[Token] = queryTokens.map {
-      case SpecialSymbol(value @ "|") => Escaped(value.charAt(0))
-      case SpecialSymbol(value @ "^") => Escaped(value.charAt(0))
-      case SpecialSymbol(value @ "$") => Escaped(value.charAt(0))
-      case SpecialSymbol(value @ "+") => Escaped(value.charAt(0))
-      case SpecialSymbol(value @ "*") => Escaped(value.charAt(0))
-      case SpecialSymbol(value @ "(") => Escaped(value.charAt(0))
-      case SpecialSymbol(value @ ")") => Escaped(value.charAt(0))
-      case SpecialSymbol(value @ ".") => Escaped(value.charAt(0))
-      case SpecialSymbol(value @ "?") => Escaped(value.charAt(0))
-      case other @ _                  => other
+      case SpecialSymbol(value) if specialChars.contains(value) => Escaped(value.charAt(0))
+      case other @ _                                            => other
     }
 
     StringAssembler.buildStringFromTokens(preciseMatch)
   }
 
-  def spaceInsenstive(query: String): String = {
-    SpaceInsensitive.spaceInsensitiveString(query)
-  }
-
   def langByLink(fileLink: String, defaultLang: String): String = {
     val ext = FilenameUtils.getExtension(fileLink)
     langByExt.getOrElse(ext, defaultLang)
   }
 
   def buildRegex(query: String, insensitive: Boolean, space: Boolean, precise: Boolean): Regex = {
-    val regex = mutable.StringBuilder.newBuilder
-    if (insensitive) {
-      regex.append("(?i)")
-    }
-    regex.append {
-      val preciseMatch = if (precise) {
-        Helper.preciseMatch(query)
-      } else {
-        query
-      }
+    val regex: String = {
+      val insensitiveCase = if (insensitive) "(?i)" else ""
+      val preciseMatch    = if (precise) Helper.preciseMatch(query) else query
+
       if (space) {
-        Helper.spaceInsenstive(preciseMatch)
+        insensitiveCase + SpaceInsensitive.spaceInsensitiveString(preciseMatch)
       } else {
-        preciseMatch
+        insensitiveCase + preciseMatch
```

Comment:
It can be done with one string concatenation.

Reply:
Will make.

Reply:
`insensitiveCase + SpaceInsensitive.spaceInsensitiveString(preciseMatch)` and `insensitiveCase + preciseMatch` functions must be removed from the `if-else` block.